### PR TITLE
Add sphinx GitHub action

### PR DESCRIPTION
**Pull request by slfuchs**
This GitHub action generates the sphinx documentation and deploys it to [PartExa/PartExa-Sphinx](https://github.com/PartExa/PartExa-Sphinx) (with GitHub page https://partexa.github.io/PartExa-Sphinx/).

### CODE REVIEW

File: .github/workflows/sphinx.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          pip install sphinx_rtd_theme
+        shell: bash
+
+      - uses: actions/checkout@v2
+      
+      - name: Compile
+        run: |
+          mkdir build
+          cd build
+          cmake ../ -DCMAKE_BUILD_TYPE=Release
```
Comment (ppraegla) [inline .github/workflows/sphinx.yml:40]: Is it possible that you need to activate the env again?

Comment (slfuchs) [reply .github/workflows/sphinx.yml:40]: Yes, it looks like we have to do that.

**Comment by slfuchs** [comment]
Note that GitHub Pages by default use Jekyll to build the site. For sphinx documentation we want to use a static site generator. We can disable Jekyll by adding a file `.nojekyll` in the `html` directory. See also description here:
https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#static-site-generators

**Comment by slfuchs** [comment]
This PR is ready for review and merge. All deploy keys are set.

**Review by ppraegla — APPROVED** [review]

**Pull request merged**